### PR TITLE
feat: add Timeline block

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes https://github.com/Codeinwp/otter-internals/issues/177

### Summary

Added the initial implementation for Timeline Block

### Screenshots 

image

image

image

----

### Test instructions

1. Inser the Timeline Block
2. Customize 
3. Check if everything is working

---- 

### Checklist before the final review

- [x] Included E2E or unit tests for the changes in this PR.
- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

File: src/blocks/blocks/timeline/group/edit.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	Spinner
+} from '@wordpress/components';
+
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+
+import { createBlock } from '@wordpress/blocks';
+
+import {
+	useSelect,
+	dispatch,
```
Comment (HardeepAsrani) [inline src/blocks/blocks/timeline/group/edit.js:30]: Seems like `dispatch` and `use` aren't being used anywhere.

File: src/blocks/blocks/timeline/group/edit.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+			<div { ...blockProps }>
+				<div className="o-timeline-root" style={inlineStyles}>
+					<InnerBlocks template={[
+						[
+							'themeisle-blocks/timeline-item',
+							{ },
+							[
+								[ 'core/paragraph', { content: 'January 15, 2024', fontSize: 'small' }],
+								[ 'core/heading', { content: 'Project Launch', level: 3 }],
```
Comment (HardeepAsrani) [inline src/blocks/blocks/timeline/group/edit.js:130]: We should still make these strings translatable.

File: src/blocks/blocks/timeline/item/edit.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+									)
+								)
+							)
+						}
+					</div>
+					<div class="o-timeline-content">
+						<InnerBlocks
+							template={
+								[
+									[ 'core/paragraph', { content: 'January 15, 2024', fontSize: 'small' }],
```
Comment (HardeepAsrani) [inline src/blocks/blocks/timeline/item/edit.js:125]: Similarly, we can make these translatable.

File: src/dashboard/components/pages/Blocks.js [hunk 4]
```diff
@@ -211,6 +213,12 @@ const otterBlocks = [
 		'name': __( 'AI Block', 'otter-blocks' ),
 		'icon': () => aiGeneration,
 		'docLink': 'https://docs.themeisle.com/article/1917-the-ai-block-otter-blocks-library#content-generator'
+	},
+	{
+		'slug': 'themeisle-blocks/timeline',
+		'name': __( 'Timeline', 'otter-blocks' ),
+		'icon': timelineIcon,
+		'docLink': 'https://docs.themeisle.com/article/2050-modal-block'
```
Comment (HardeepAsrani) [inline src/dashboard/components/pages/Blocks.js:221]: You can draft an article on the HS docs (or ask someone from Support team for help) and put that URL here.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Looks good, left few small things here and there. Good work, the block works great.

File: src/blocks/blocks/timeline/item/block.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	"textdomain": "otter-blocks",
+	"attributes": {
+		"id": {
+			"type": "string"
+		},
+		"containerBorder": {
+			"type": "object"
+		},
+		"containerRadius": {
+			"type": ["string", "object"]
```
[the remark this answers is not in the document]
Comment (Soare-Robert-Daniel) [reply src/blocks/blocks/timeline/item/block.json:19]: I tried to use the new border radius component from Gutenberg, but was it not available to use. That is its signature

File: src/dashboard/components/pages/Blocks.js (hunk 4 above)
Comment (ineagu) [reply src/dashboard/components/pages/Blocks.js]: it's done by me. is in draft.

**Review by HardeepAsrani — APPROVED** [review]

**Comment by carlalexander** [comment]
@Soare-Robert-Daniel didn't get to review before the merge, but I'd have merged the `test` commit into the `feat` one since it's a test for that feature.

**Pull request merged**